app/v2/notifications/rest_push.py

```diff
@@ -1,59 +1,99 @@
+    return push_notification_helper(push_notification_request)
 
 
 @v2_notification_blueprint.route('/push/broadcast', methods=['POST'])
 def send_push_broadcast_notification():
-    return push_notification_helper(push_notification_broadcast_request, True)
+    return push_notification_helper(push_notification_broadcast_request)
 
 
-def push_notification_helper(schema: dict, is_broadcast: bool):
+def push_notification_helper(schema: dict):
     """
     Note that this helper cannot be called other than as part of a request because it accesses
     the Flask "request" instance.
     """
-
-    if not is_feature_enabled(FeatureFlag.PUSH_NOTIFICATIONS_ENABLED):
-        raise NotImplementedError()
-
     if not authenticated_service.has_permissions(PUSH_TYPE):
         raise BadRequestError(
-            message='Service is not allowed to send {}'.format(get_public_notify_type_text(PUSH_TYPE, plural=True))
+            message='Service is not allowed to send {}'.format(
+                get_public_notify_type_text(PUSH_TYPE, plural=True),
+            ),
+            status_code=403,
         )
 
-    req_json = validate(request.get_json(), schema)
+    payload = validate_push_payload(schema)
 
-    if 'mobile_app' in req_json:
-        app_instance = mobile_app_registry.get_app(MobileAppType[req_json['mobile_app']])
+    try:
+        # Choosing to use the email queue for push to limit the number of empty queues
+        deliver_push.delay(payload)
+    except (CeleryError, OperationalError):
+        current_app.logger.exception('Failed to enqueue deliver_push request')
+        response = jsonify(result='error', message='VA Notify service impaired, please try again'), 502
     else:
-        app_instance = mobile_app_registry.get_app(DEAFULT_MOBILE_APP_TYPE)
+        response = jsonify(result='success'), 201
+    # Flask turns the tuple into a json and status_code
+    return response
+
+
+def validate_push_payload(schema: dict[str, str]) -> V2PushPayload:
+    """Validate an incoming push request.
+
+    Args:
+        schema (dict[str, str]): The incoming request
 
+    Raises:
+        BadRequestError: Failed validation
+
+    Returns:
+        dict[str, str]: Validated request dictionary
+    """
     try:
-        vetext_client.send_push_notification(
-            app_instance.sid,
-            req_json['template_id'],
-            req_json['topic_sid'] if is_broadcast else req_json['recipient_identifier']['id_value'],
-            is_broadcast,
-            req_json.get('personalisation'),
-        )
-    except VETextBadRequestException as e:
-        raise BadRequestError(message=e.message, status_code=400)
-    except (VETextNonRetryableException, VETextRetryableException):
-        return jsonify(result='error', message='Invalid response from downstream service'), 502
-    else:
-        return jsonify(result='success'), 201
+        req_json: dict[str, str] = validate(request.get_json(), schema)
```

Comment:
We generally need to stop writing these long, Java-esq try/except blocks.  It's not PEP8 compliant.  This line might raise ValidationError, and it is the only line that should be in this "try" block.  The subsequent if/else should not raise any exception.  (Please verify that, if necessary.)

Reply:
Ok - I believe we can remove the TypeError. 

I believe this still needs to stay in the Try/Except block. The `get_app` method was updated to throw a KeyError we may need to catch and log.

Reply:
Yes, that's appropriate if get_app can raise KeyError.

Reply:
It should still be broken up.  Similar to this . . .
```python
try:
  req_json = validate(...)
except ValidationError:
  pass

try:
  app_sid = ...
except KeyError:
  pass
```

Reply:
Ah gotcha. I like that

Reply:
Have a specific section of the pep8 for me @kalbfled ?

Reply:
Updated but would also appreciate seeing an reference @kalbfled

Reply:
Attempted to update this which lead down a rabbit hole - Reverted. 

https://github.com/department-of-veterans-affairs/notification-api/pull/2266#discussion_r1939893772

Reply:
From PEP8: "Additionally, for all try/except clauses, limit the try clause to the absolute minimum amount of code necessary. Again, this avoids masking bugs."

Reply:
The link to that in PEP8 (scroll down or just search for the sentence) is https://peps.python.org/pep-0008/#programming-recommendations

Reply:
Thank you - I will keep this in mind for the future. At this time - I would prefer not to refactor this unless its a requirement.